src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs

```diff
@@ -14,84 +14,86 @@
-				return false;
-			}
-
-			if (enableLLVM && (ndkVersion.Version < 10 || (ndkVersion.Version == 10 && ndkVersion.Revision[0] < 'd'))) {
-				log.LogCodedError ("XA3005",
-						"The detected Android NDK version is incompatible with the targeted LLVM configuration, " +
-						"please upgrade to NDK r10d or newer.");
+						"The detected Android NDK version is incompatible with this version of Xamarin.Android, " +
+						"please upgrade to NDK r19 or newer.");
 			}
 
 			return true;
 		}
 
-		public static string GetNdkToolPrefix (string androidNdkPath, AndroidTargetArch arch)
+		public static string GetNdkToolPrefix (string androidNdkPath, AndroidTargetArch arch, int apiLevel)
 		{
-			var path = GetNdkTool (androidNdkPath, arch, "as");
+			if (!UsingNewNDK)
+				return NdkUtilOld.GetNdkToolPrefix (androidNdkPath, arch);
+
+			var path = GetNdkTool (androidNdkPath, arch, "as", apiLevel);
 			if (path != null)
 				path = path.Substring (0, path.LastIndexOf ("-") + 1);
 			return path;
 		}
 
-		public static List<string> GetNdkToolchainPath(string androidNdkPath, AndroidTargetArch arch)
+		public static string GetNdkTool (string androidNdkPath, AndroidTargetArch arch, string tool, int apiLevel)
 		{
-			var toolchains  = GetNdkToolchainDirectories (Path.Combine (androidNdkPath, "toolchains"), arch);
-			if (!toolchains.Any ())
-				Diagnostic.Error (5101,
-						"Toolchain directory for target {0} was not found.",
-						arch);
-			// Sort the toolchains paths in reverse so that we prefer the latest versions.
-			Array.Sort(toolchains);
-			Array.Reverse(toolchains);
-
-			return new List<string>(toolchains);
-		}
-
-		public static string GetNdkTool (string androidNdkPath, AndroidTargetArch arch, string tool)
-		{
-			var toolchains = GetNdkToolchainPath(androidNdkPath, arch);
-			string extension = OS.IsWindows ? ".exe" : string.Empty;
-			List<string> toolPaths  = null;
-			foreach (var platbase in toolchains) {
-				string path = Path.Combine (platbase, "prebuilt", MonoAndroidHelper.AndroidSdk.AndroidNdkHostPlatform, "bin", GetNdkToolchainPrefix (arch) + tool + extension);
-				if (File.Exists (path))
-					return path;
-				if (toolPaths == null)
-					toolPaths = new List<string>();
-				toolPaths.Add (path);
-			}
-			{
-				string path = Path.Combine (androidNdkPath, "prebuilt", MonoAndroidHelper.AndroidSdk.AndroidNdkHostPlatform, "bin", tool);
-				if (File.Exists (path))
-					return path;
-				if (toolPaths == null)
-					toolPaths = new List<string>();
-				toolPaths.Add (path);
-                       }
+			if (!UsingNewNDK)
+				return NdkUtilOld.GetNdkTool (androidNdkPath, arch, tool);
+
+			string toolchainDir = Path.Combine (androidNdkPath, "toolchains", "llvm", "prebuilt", MonoAndroidHelper.AndroidSdk.AndroidNdkHostPlatform);
+			string toolName;
+			bool forCompiler = false;
+
+			if (String.Compare (tool, "gcc", StringComparison.Ordinal) == 0 ||
+			    String.Compare (tool, "clang", StringComparison.Ordinal) == 0) {
```

Comment:
Code formatting style: indent primarily with tabs, and this expression should be indented *more* than the following block, e.g.

```csharp
if (String.Compare (tool, "gcc", StringComparison.Ordinal) == 0 ||
        String.Compare (tool, "clang", StringComparison.Ordinal) == 0) {
    forCompiler = true;
    toolName = "clang";
}
```